packages/edge-config/src/utils/create-loaders.ts

```diff
@@ -0,0 +1,486 @@
+import DataLoader from 'dataloader';
+import { readFile } from '@vercel/edge-config-fs';
+import type {
+  Connection,
+  EdgeConfigItems,
+  EdgeConfigValue,
+  EmbeddedEdgeConfig,
+} from '../types';
+import { fetchWithHttpCache } from './fetch-with-http-cache';
+import { measure, trace } from './tracing';
+import { ERRORS, hasOwnProperty, isDynamicServerError } from '.';
+
+const jsonParseCache = new Map<string, unknown>();
+
+const readFileTraced = trace(readFile, { name: 'readFile ' });
+const jsonParseTraced = trace(JSON.parse, { name: 'JSON.parse' });
+const cachedJsonParseTraced = trace(
+  (edgeConfigId: string, content: string) => {
+    const cached = jsonParseCache.get(edgeConfigId);
+    if (cached) return cached;
+
+    const parsed = jsonParseTraced(content) as unknown;
+
+    // freeze the object to avoid mutations of the return value of a "get" call
+    // from affecting the return value of future "get" calls
+    jsonParseCache.set(edgeConfigId, Object.freeze(parsed));
```

Comment:
That wouldn't prevent nested objects from being changed, right? So we'd need to recursively freeze it.